src/rand.jl

```diff
@@ -0,0 +1,48 @@
+import Random
+
+# TODO: remove Random.CloseOpen01{BigFloat} parameters
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    st::Random.SamplerType{TOrRef},
+    n::Random.Repetition,
+) where {TOrRef<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
+
+    T = _nonreftype(TOrRef)
+    return Random.SamplerSimple(
+        Random.SamplerType{T}(),
+        Random.SamplerBigFloat{Random.CloseOpen01{BigFloat}}(precision(T))
+    )
+end
+
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arf}) where {A,B} =
+    Arf(rand(rng, sp.data), prec = sp.data.prec)
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arb}) where {A,B} =
+    Arb(rand(rng, sp.data), prec = sp.data.prec)
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Acb}) where {A,B} =
+    Acb(rand(rng, sp.data), rand(rng, sp.data), prec = sp.data.prec)
+
+function Random.rand(
+    rng::Random.AbstractRNG,
+    st::Random.SamplerTrivial{A,A},
+) where {A<:ArfOrRef}
+    x = rand(rng, Random.SamplerBigFloat{Random.CloseOpen01{BigFloat}}(precision(st[])))
```

Comment:
This is equivalent as what you had before; what I meant is that you could store the `SamplerBigFloat` in a `SamplerSimple` like you do for the types version (`rand(Arb)`). Again, this won't be a huge improvement, but will manifest for example when creating arrays.

Reply:
I see, you meant creating `SamplerSimple` instead of relying on the automatic creation of `SamplerTrivial`.